evcxr/src/eval_context.rs

```diff
@@ -508,6 +508,31 @@ impl EvalContext {
         Ok(completions)
     }
 
+    pub fn hover(
+        &mut self,
+        code: &str,
+        state: &mut ContextState,
+    ) -> Result<String> {
+        let (modified_code, hover_offset) = if code == "let" {
+            (String::from("let _ = 1;"), 0)
+        } else {
+            (format!("{};", code), code.len() as u32)
+        };
+        let (user_code, code_info) = CodeBlock::from_original_user_code(&modified_code);
+        let user_code = state.apply(user_code, &code_info.nodes)?;
+        let pad_code = state.analysis_code(user_code);
+        self.analyzer.set_source(pad_code.code_string())?;
+        //if I set the hover position to be the start position of input code, then 
```

Comment:
That makes sense and is consistent with what I observe if I hover over the "std" in vscode